vllm_detector_adapter/generative_detectors/base.py

```diff
@@ -66,16 +70,16 @@ def load_template(self, template_path: Optional[Union[Path, str]]) -> str:
         logger.info("Using supplied template:\n%s", resolved_template)
         return self.jinja_env.from_string(resolved_template)
 
-    def apply_task_template(
+    def apply_task_template_to_chat(
         self, request: ChatDetectionRequest
     ) -> Union[ChatDetectionRequest, ErrorResponse]:
-        """Apply task template on the request"""
+        """Apply task template on the chat request"""
         return request
 
-    def preprocess(
+    def preprocess_chat_request(
```

Comment:
feels a bit weird that `/chat` has its own preprocess function, I am assuming this is for llama guard but not for context.